future_webpage/dgf/cronjobs.py

```diff
@@ -1,13 +1,32 @@
 import logging
+from decimal import Decimal
 
 from .models import Friend
 from .pdga import PdgaApi
 
 logger = logging.getLogger(__name__)
 
 
-def fetch_rating():
-    pdga_service = PdgaApi()
+def update_tournaments(pdga_service):
+    for friend in Friend.objects.all():
+        if friend.pdga_number:
+            statistics = pdga_service.query_player_statistics(pdga_number=friend.pdga_number)
+
+            money_earned = 0
+            tournaments = 0
+            for yearly_stats in statistics['players']:
+                try:
+                    money_earned += Decimal(yearly_stats['prize'])
```

Comment:
sabes cómo hacer esto en una lambda con el try catch? es que si la key 'prize' no está presente pega un pete en vez de saltarselo... 

no se si queda más leíble así o añadiendo un algo como '[x for i in statistics['players'] if has_key['prize']: x]' o algo así extraño, creo que es menos leible... pero bueno... qué opinas?

Reply:
Tu lo que quieres es algo así:

```python
money_earned = sum([Decimal(yearly_stats['price']) for p in statistics['players'] if 'price' in yearly_stats])
```

no?

yo lo veo legible

Reply:
o pa escribirlo más bonito:

```python
money_earned = sum(earned_money(yearly_stat) for yearly_stat in statistics['players']])

earned_money = lambda x: Decimal(x['price'] if 'price' in x else 0

```

Reply:
BLAS... si te mola más, lo hago

Reply:
Pero con la iteración de lo otro hago dos cosas en una iteración... no? es mejor un sólo for... 
también estoy pensando en extenderlo en el futuro con más datos...

Reply:
po si... si al final lo exiendes hacer un blas tampoco mola

Reply:
por cierto, a lo mejor tiene sentido hacer un método aparte que sea "rellenar un Friend con los datos de la PDGA"

Reply:
porque ahor amismo estamos en un for-if-for-try y en el futuro cuando se amplie tendrá antes o despues o dentro de ese `try` algún que otro if, un for, etc..

Reply:
sip, creo que eso vendrá en la próxima iteración :)